crates/rpc/src/methods/token_add.rs

```diff
@@ -0,0 +1,317 @@
+use crate::{Error, Result};
+use ethui_dialogs::{Dialog, DialogMsg};
+use ethui_networks::Networks;
+use ethui_types::{Address, GlobalState, TokenMetadata, U256};
+use ethui_wallets::{WalletControl, Wallets};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug)]
+pub struct TokenAdd {
+    erc20_token: Option<TokenMetadata>,
+    erc721_token: Option<ERC721Data>,
+    erc1155_token: Option<ERC1155Data>,
+    chain_id: Option<u32>,
+    _type: String,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct ERC721Data {
+    address: Address,
+    token_id: U256,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+struct ERC1155Data {
+    address: Address,
+    token_id: U256,
+}
+
+impl TokenAdd {
+    pub fn build() -> TokenAddBuilder {
+        TokenAddBuilder::default()
+    }
+
+    pub async fn get_current_chain_id(&self) -> Result<u32> {
+        let networks = Networks::read().await;
+        let current_network = networks.get_current();
+        Ok(current_network.chain_id)
+    }
```

Comment:
this whole function never panics, so `Result` seems useless here. seems you can directly return `u32`